packages/live-share-inking/src/core/InkingManager.ts

```diff
@@ -0,0 +1,1293 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the Microsoft Live Share SDK License.
+ */
+
+import { EventEmitter } from "events";
+import { CanvasReferencePoint, InkingCanvas } from "../canvas/InkingCanvas";
+import { DryCanvas, WetCanvas } from "../canvas/DryWetCanvas";
+import { LaserPointerCanvas } from "../canvas/LaserPointerCanvas";
+import { IPoint, IPointerPoint, IRect, makeRectangle, screenToViewport, viewportToScreen } from "./Geometry";
+import { Stroke, IStroke, IStrokeCreationOptions, StrokeType } from "./Stroke";
+import { InputFilter, InputFilterCollection } from "../input/InputFilter";
+import { JitterFilter } from "../input/JitterFilter";
+import { generateUniqueId, pointerEventToPoint } from "./Utils";
+import { InputProvider } from "../input/InputProvider";
+import { PointerInputProvider } from "../input/PointerInputProvider";
+import { DefaultHighlighterBrush, DefaultLaserPointerBrush, DefaultPenBrush, IBrush } from "./Brush";
+
+interface CoalescedPointerEvent extends PointerEvent {
+    getCoalescedEvents(): PointerEvent[];
+}
+
+/**
+ * @hidden
+ * Retrieves a list of coalesced PointerEvent objects from a single PointerEvent.
+ * PointerEvent.getCoalescedEvents() is an experimental feature and TypeScript doesn't
+ * yet declare it.
+ * For more info: https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/getCoalescedEvents
+ * @param event The PointerEvent to get coalesced events from.
+ * @returns An array of coalesced PointerEvent objects.
+ */
+function getCoalescedEvents(event: PointerEvent): PointerEvent[] {
+    if ('getCoalescedEvents' in event) {
+        const events: PointerEvent[] = (event as CoalescedPointerEvent).getCoalescedEvents();
+
+        // Firefox can return an empty list. See https://bugzilla.mozilla.org/show_bug.cgi?id=1511231.
+        if (events.length >= 1) {
+            return events;
+        }
+    }
+
+    return [event];
+}
+
+/**
+ * Defines available inking tools.
+ */
+export enum InkingTool {
+    Pen,
+    LaserPointer,
+    Highlighter,
+    Eraser,
+    PointEraser
+}
+
+type StrokeBasedTool = InkingTool.Pen | InkingTool.LaserPointer | InkingTool.Highlighter;
+
+/**
+ * The event emitted by InkingManager when the canvas is cleared.
+ */
+export const ClearEvent: symbol = Symbol();
```

Comment:
>ClearEvent: symbol = Symbol(); [](http://example.com/codeflow?start=13&length=30)

Is there a significant advantage to having these be symbols versus just strings? All of the other events in the SDK are strings which is why I ask... I haven't looked at where these are consumed yet

Reply:
I borrowed using symbols from the WInk codebase. Symbols are guaranteed to be unique.

I'll change to strings for consistency.

Reply:
Fixed in https://github.com/microsoft/live-share-sdk/commit/18b13a631f0c2833a77b7d5181cb89b36206c512